### DISCUSSION TITLE
Lorentzian 3D Levi-Civita

### DISCUSSION THREAD

**Discussion by nmoynihan** [FeynCalc/feyncalc#95] [Q&A]
I want to evaluate quantities such as LeviCivita[a,b,c] in Feyncalc for a 2+1 dimensional spacetime. Is there a command for this? In 4D, we can use e.g. LC[a, b, c, d] LC[a, b, c, e] // Contract = -6MT[d,e]. However, in 2+1 dimensions this doesn't work.

Is there an equivalent of LC[a, b, c] LC[a, b, e] // Contract which would give me something proportional to the metric tensor (rather than the Kroneker Delta, like CLCD)?

**Comment by vsht** [comment]
There is `CLC[a, b, c]`, but it is the usual Cartesian 3D Levi-Civita embedded in the 4-dimensional space-time.
As discussed in

https://github.com/FeynCalc/feyncalc/discussions/61#discussioncomment-50037

and

https://github.com/FeynCalc/feyncalc/discussions/63

it is currently not possible to have anything different from 3+1 (and it would require considerable effort to change that).

**Comment by nmoynihan** [comment]
OK, thanks.

As an ugly hack, which can probably be improved upon, a simple way to do this is to just work in 4D and define a compactification vector `n' which is orthogonal to all other momenta/polarizations etc. Then define a 3D Levi-Civita and special contraction given by

`LC3D[a_, b_, c_] := 
 LC[a, b, c, d] FV[n, d]; 
ScalarProduct[n, n] = 1;
LC3DContract[f_] := (f // Contract) //. Pair[Momentum[n], x_] :> 0;`
